src/flamenco/vm/syscall/fd_vm_syscall_util.c

```diff
@@ -17,38 +18,58 @@ fd_vm_syscall_abort( FD_PARAM_UNUSED void *  _vm,
                      FD_PARAM_UNUSED ulong   r4,
                      FD_PARAM_UNUSED ulong   r5,
                      FD_PARAM_UNUSED ulong * _ret ) {
+  /* https://github.com/anza-xyz/agave/blob/v2.0.6/programs/bpf_loader/src/syscalls/mod.rs#L630 */
+  return FD_VM_ERR_ABORT;
+}
 
-  /* FIXME: this originally cleared *_ret, which would change r0 to 0 as
-     part of the abort.  This is commented out below to preserve VM
-     state at precisely the time of the abort (including the updaets to
-     ic and cu for the syscall itself).  It is trivial to flip back if
-     desired by uncommenting the below. */
+/* FD_TRANSLATE_STRING returns a read only pointer to the host address of
+   a valid utf8 string, or it errors.
 
-//*_ret = 0;
+   Analogous of Agave's translate_string_and_do().
+   https://github.com/anza-xyz/agave/blob/v2.0.6/programs/bpf_loader/src/syscalls/mod.rs#L601
 
-  return FD_VM_ERR_ABORT;
-}
+   As of v0.2.6, the only two usages are in syscall panic and syscall log. */
+#define FD_TRANSLATE_STRING( vm, vaddr, msg_sz ) (__extension__({          \
+    char const * msg = FD_VM_MEM_SLICE_HADDR_LD( vm, vaddr, 1UL, msg_sz ); \
+    if( FD_UNLIKELY( !fd_utf8_verify( msg, msg_sz ) ) ) {                  \
+      return FD_VM_ERR_INVAL; /* Err(SyscallError::InvalidString(...)) */  \
+    }                                                                      \
+    msg;                                                                   \
+}))
```

Comment:
Be aware that if msg_sz is zero than msg is an undefined pointer (as described in the documentation of FD_VM_MEM_SLICE_HADDR_LD).

Reply:
That's ok, `fd_utf8_verify` works with msg_sz==0.

Reply:
```rust
  char const * file = FD_TRANSLATE_STRING( vm, file_vaddr, file_sz );

  /* Max msg_sz: 37 - 6 + 20 + 20 + file_sz = 71 + file_sz <= 127 => file_sz <= 56 */
  if( file_sz<=56 ) {
    fd_log_collector_printf_dangerous_max_127( vm->instr_ctx,
      "SBF program Panicked in %s at %lu:%lu", file, line, column );
  }
  ```

In the case where file_sz==0, we pass a potentially undefined pointer to fd_log_collector_printf_dangerous_max_127 - is this safe?

Reply:
Oh right - fixed with "%.*s", run some tests to confirm. We definitely need some test vectors for panic.